sdk/go/common/util/env/env.go

```diff
@@ -212,12 +279,29 @@ func (v value) Var() Var {
 	return v.variable
 }
 
+// Get the value that will be used.
+func (v value) get() string {
+	defValue := v.variable.options.defaultValue
```

Comment:
Just thinking out loud:
Would it be preferable to compute the default value here by calling the function passed to `DefaultF()`?

That way, the restriction that the variable fails validation if DefaultF failed comes into play only if the default was actually needed. That is, if DefaultF would have failed (if we used it), but we don't use it because a value was set for that env var, the variable will still pass validation.